pkg/cidata/cidata.go

```diff
@@ -50,17 +51,23 @@ func setupEnv(y *limayaml.LimaYAML) (map[string]string, error) {
 			}
 		}
 	}
-	// Replace "localhost" in proxy settings with the gateway address
-	localhostRegexes := []*regexp.Regexp{
-		regexp.MustCompile(`\blocalhost\b`),
-		regexp.MustCompile(`\b127.0.0.1\b`),
-	}
+	// Replace IP that IsLoopback in proxy settings with the gateway address
 	for _, name := range append(lowerVars, upperVars...) {
 		value, ok := env[name]
 		if ok && !strings.EqualFold(name, "no_proxy") {
-			for _, re := range localhostRegexes {
-				value = re.ReplaceAllString(value, qemu.SlirpGateway)
+			u, err := url.Parse(value)
+			if err != nil {
+				logrus.Warnf("Ignoring invalid proxy %q=%v: %s", name, value, err)
+				continue
+			}
+
+			ips, _ := net.LookupIP(u.Hostname())
+			for _, ip := range ips {
+				if ip.IsLoopback() {
+					u.Host = net.JoinHostPort(qemu.SlirpGateway, u.Port())
```

Comment:
Looks like `net.JoinHostPort("127.0.0.1", "")` is buggy and returns `127.0.0.1:` instead of `127.0.0.1`.

So you need to check if `u.Port()` returns the empty string.

The test cases should include at least one host without a port.